toolchain/check/handle_impl.cpp

```diff
@@ -56,23 +69,57 @@ auto HandleParseNode(Context& context, Parse::ForallId /*node_id*/) -> bool {
 
 auto HandleParseNode(Context& context, Parse::ImplTypeAsId node_id) -> bool {
   auto [self_node, self_id] = context.node_stack().PopExprWithNodeId();
-  auto self_type_inst_id = ExprAsType(context, self_node, self_id).inst_id;
-  context.node_stack().Push(node_id, self_type_inst_id);
+  auto self_type = ExprAsType(context, self_node, self_id);
+
+  const auto& introducer = context.decl_introducer_state_stack().innermost();
+  if (introducer.modifier_set.HasAnyOf(KeywordModifierSet::Extend)) {
+    auto class_scope =
+        TryAsClassScope(context, context.decl_name_stack().PeekParentScopeId());
+    if (class_scope) {
```

Comment:
I think the mixin comment should be here instead of inside `TryAsClassScope`.
```suggestion
    auto class_scope =
        TryAsClassScope(context, context.decl_name_stack().PeekParentScopeId());
    // TODO: This should also be done in a mixin.
    if (class_scope) {
```

Reply:
Done